bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericMetadataProvider.java

```diff
@@ -69,15 +96,21 @@ public void addMetadata(String bindingType, String itemName, String value,
      * @param itemName
      */
     public void removeMetadata(String itemName) {
-        Set<Metadata> toBeRemoved;
+        Set<MetadataKey> toBeRemoved;
+        Set<Metadata> removed = new HashSet<>();
         try {
             lock.writeLock().lock();
-            toBeRemoved = metadata.stream().filter(MetadataPredicates.ofItem(itemName)).collect(toSet());
-            metadata.removeAll(toBeRemoved);
+            toBeRemoved = metadata.keySet().stream() //
+                    .filter(key -> key.getItemName().equals(itemName)) //
+                    .filter(key -> !key.getNamespace().startsWith(MetadataRegistry.INTERNAL_NAMESPACE_PREFIX)) //
```

Comment:
So you filter out internal metadata. `removeMetadata` will then change its behavior and NOT delete internal metadata. Is this the intention? How are we supposed to delete internal metadata then?

Reply:
> How are we supposed to delete internal metadata then?

That's the whole point: _WE_ aren't. It's internal. And therefore it should be kept something internal to the framework.

Reply:
I got your point but I see these commands as some useful tool a developer needs. Therefore I also added the `smarthome metadata listinternal` command. There may be times where it comes handy for a developer to remove even internal metadata so it would be nice to have this feature.